javaparser-symbol-solver-testing/src/test/test_sourcecode/javaparser_new_src/javaparser-core/com/github/javaparser/ast/validator/TypedValidator.java

```diff
@@ -0,0 +1,24 @@
+package com.github.javaparser.ast.validator;
+
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ast.Node;
+
+import java.util.function.BiConsumer;
+
+/**
+ * A validator that validates a known node type.
+ */
+public interface TypedValidator<N extends Node> extends BiConsumer<N, ProblemReporter> {
+    /**
+     * @param node the node that wants to be validated
+     * @param problemReporter when found, validation errors can be reported here
```

Comment:
@ftomassetti is it okay to add new files in this directory?

Reply:
No, it should basically be inserted anywhere else but here :)

Reply:
Of course also the Validator.java file should be moved. I would create a directory under the root of the resources folder and put both files there